build-logic/src/main/kotlin/polaris-java.gradle.kts

```diff
@@ -30,18 +30,34 @@ plugins {
   `java-library`
   `java-test-fixtures`
   `jvm-test-suite`
+  checkstyle
   id("polaris-spotless")
   id("jacoco-report-aggregation")
   id("net.ltgt.errorprone")
 }
 
 apply<PublishingHelperPlugin>()
 
-if (project.extra.has("duplicated-project-sources")) {
-  // skip the style check for duplicated projects
-  tasks.withType<Checkstyle>().configureEach { enabled = false }
+val checkstyleVersion =
+  versionCatalogs
+    .named("libs")
+    .findVersion("checkstyle")
+    .orElseThrow { GradleException("checkstyle version not found in libs.versions.toml") }
+    .requiredVersion
+
+checkstyle {
+  toolVersion = checkstyleVersion
+  configFile = rootProject.file("codestyle/checkstyle.xml")
+  isIgnoreFailures = false
+  maxErrors = 0
+  maxWarnings = 0
 }
 
+// Ensure Checkstyle runs after jandex to avoid task dependency issues
+tasks
+  .matching { it.name.startsWith("checkstyle") }
```

Comment:
This is a trap and prevents "configure on demand", `.matching` gives you the fully configured tasks.
I'd use
```kotlin
tasks.withType<Checkstyle>().configureEach { ...
```